linera-views/src/views/set_view.rs

```diff
@@ -939,70 +939,78 @@ pub type HashedCustomSetView<C, I> =
 mod graphql {
     use std::borrow::Cow;
 
+    use serde::{de::DeserializeOwned, Serialize};
+
     use super::{CustomSetView, SetView};
-    use crate::context::Context;
+    use crate::{
+        common::CustomSerialize,
+        context::Context,
+        graphql::{hash_name, mangle},
+    };
 
-    impl<C: Context, I: async_graphql::OutputType> async_graphql::OutputType for SetView<C, I>
-    where
-        I: serde::ser::Serialize + serde::de::DeserializeOwned + Clone + Send + Sync,
-    {
+    impl<C: Send + Sync, I: async_graphql::OutputType> async_graphql::TypeName for SetView<C, I> {
         fn type_name() -> Cow<'static, str> {
-            format!("[{}]", I::qualified_type_name()).into()
+            format!(
+                "SetView_{}_{:08x}",
+                mangle(I::type_name()),
+                hash_name::<I>(),
+            )
+            .into()
         }
+    }
 
-        fn qualified_type_name() -> String {
-            format!("[{}]!", I::qualified_type_name())
+    #[async_graphql::Object(cache_control(no_cache), name_type)]
+    impl<C, I> SetView<C, I>
+    where
+        C: Context,
+        I: Send + Sync + Serialize + DeserializeOwned + async_graphql::OutputType,
+    {
+        async fn elements(&self, count: Option<usize>) -> Result<Vec<I>, async_graphql::Error> {
+            let mut indices = self.indices().await?;
+            Ok(if let Some(count) = count {
+                indices.truncate(count);
+                indices
+            } else {
+                indices
+            })
```

Comment:
(Maybe more concise; also below.)

```suggestion
            let mut indices = self.indices().await?;
            if let Some(count) = count {
                indices.truncate(count);
            }
            Ok(indices)
```